packages/flutter/test/material/slider_test.dart

```diff
@@ -3396,6 +3396,117 @@ void main() {
     expect(dragStarted, false);
   });
 
+  testWidgetsWithLeakTracking('Slider does not request focus when the value is changed', (WidgetTester tester) async {
+    double value = 0.5;
+    const Color overlayColor = Color(0xffff0000);
+
+    Widget buildApp({bool enabled = true}) {
+      return MaterialApp(
+        home: Material(
+          child: Center(
+            child: StatefulBuilder(builder: (BuildContext context, StateSetter setState) {
+              return Slider(
+                value: value,
+                overlayColor: const MaterialStatePropertyAll<Color?>(overlayColor),
+                onChanged: enabled
+                  ? (double newValue) {
+                      setState(() {
+                        value = newValue;
+                      });
+                    }
+                  : null,
+              );
+            }),
+          ),
+        ),
+      );
+    }
+    await tester.pumpWidget(buildApp());
+
+    // Slider does not have focus when enabled and not tapped.
+    await tester.pumpAndSettle();
+    // ignore: avoid_dynamic_calls, invalid_assignment
+    final FocusNode focusNode = (tester.firstState(find.byType(Slider)) as dynamic).focusNode;
+    expect(value, equals(0.5));
+    expect(focusNode.hasFocus, false);
+
+    final Offset sliderCenter = tester.getCenter(find.byType(Slider));
+    final Offset tapLocation = Offset(sliderCenter.dx + 50, sliderCenter.dy);
+
+    // Tap somewhere to change the value.
+    final TestGesture gesture = await tester.createGesture();
+    await gesture.addPointer();
+    await gesture.down(tapLocation);
+    await gesture.up();
+    await tester.pumpAndSettle();
+    expect(value, isNot(equals(0.5)));
+    expect(focusNode.hasFocus, false);
+  });
```

Comment:
Based on your suggestions I changed the following also:

https://github.com/flutter/flutter/pull/139021/commits/ee15d3cd8cb3ebb0998babe07c33098328fcdce6

1. Add comment for regression test.
2. Add await for `tester.pumpAndSettle()`.
3. Fix the typo.
4. Remove the kind parameter of the gesture because I think it's redundant.
5. Change the FocusNode usage because it only occurs on auto-generated FocusNode.